src/Compilers/Core/Portable/Operations/OperationCloner.cs

```diff
@@ -531,11 +531,18 @@ public override IOperation VisitConstantPattern(IConstantPatternOperation operat
         public override IOperation VisitDeclarationPattern(IDeclarationPatternOperation operation, object argument)
         {
             return new DeclarationPatternOperation(
-                operation.InputType, operation.MatchedType, operation.DeclaredSymbol, operation.MatchesNull,
-                ((Operation)operation).OwningSemanticModel, operation.Syntax, operation.IsImplicit);
+                operation.MatchedType,
+                operation.MatchesNull,
+                operation.DeclaredSymbol,
+                operation.InputType,
+                ((Operation)operation).OwningSemanticModel,
+                operation.Syntax,
+                operation.Type,
+                operation.ConstantValue,
+                operation.IsImplicit);
         }
 
-        internal override IOperation VisitRecursivePattern(IRecursivePatternOperation operation, object argument)
+        public override IOperation VisitRecursivePattern(IRecursivePatternOperation operation, object argument)
```

Comment:
>public [](start = 8, length = 6)

it looks like these were added to PublicAPI.txt. Were they just inadvertently left internal before?

Reply:
Yep.